packages/shell-api/src/field-level-encryption.spec.ts

```diff
@@ -366,75 +365,45 @@ describe('Field Level Encryption', () => {
+      it('calls addKeyAltName on libmongocrypt', async() => {
         const r = { value: { ok: 1 } } as any;
-        sp.findOneAndUpdate.resolves(r);
+        libmongoc.addKeyAltName.resolves(r.value);
         const result = await keyVault.addKeyAlternateName(KEY_ID, 'altname');
-        expect(sp.findOneAndUpdate).to.have.been.calledOnceWithExactly(
-          DB,
-          COLL,
-          { _id: KEY_ID },
-          { $addToSet: { 'keyAltNames': 'altname' }, $currentDate: { 'updateDate': true } },
-          { returnDocument: 'before' }
+        expect(libmongoc.addKeyAltName).to.have.been.calledOnceWithExactly(
+          KEY_ID,
+          'altname'
         );
-        expect(result).to.deep.equal({ ok: 1 });
+        expect(result).to.deep.equal(r.value);
       });
     });
     describe('removeKeyAlternateName', () => {
-      it('calls findOneAndUpdate on key coll without empty result', async() => {
-        const r = { value: { ok: 1, keyAltNames: ['other'] } } as any;
-        sp.findOneAndUpdate.resolves(r);
+      it('calls removeKeyAltName on libmongocrypt', async() => {
+        const r = { value: { ok: 1 } } as any;
+        libmongoc.removeKeyAltName.resolves(r.value);
         const result = await keyVault.removeKeyAlternateName(KEY_ID, 'altname');
-        expect(sp.findOneAndUpdate).to.have.been.calledOnceWithExactly(
-          DB,
-          COLL,
-          { _id: KEY_ID },
-          { $pull: { 'keyAltNames': 'altname' }, $currentDate: { 'updateDate': true } },
-          { returnDocument: 'before' }
+        expect(libmongoc.removeKeyAltName).to.have.been.calledOnceWithExactly(
+          KEY_ID,
+          'altname'
         );
-        expect(result).to.deep.equal({ ok: 1, keyAltNames: ['other'] });
-      });
-      it('calls findOneAndUpdate on key coll with empty result', async() => {
-        const r = { value: { ok: 1, keyAltNames: ['altname'] } } as any;
-        const r2 = { value: { ok: 2 } } as any;
-        sp.findOneAndUpdate.onFirstCall().resolves(r);
-        sp.findOneAndUpdate.onSecondCall().resolves(r2);
-        const result = await keyVault.removeKeyAlternateName(KEY_ID, 'altname');
-        const calls = sp.findOneAndUpdate.getCalls();
-        expect(calls.length).to.equal(2);
-        expect(calls[0].args).to.deep.equal([
-          DB,
-          COLL,
-          { _id: KEY_ID },
-          { $pull: { 'keyAltNames': 'altname' }, $currentDate: { 'updateDate': true } },
-          { returnDocument: 'before' }
-        ]);
-        expect(calls[1].args).to.deep.equal([
-          DB,
-          COLL,
-          { _id: KEY_ID, keyAltNames: { $size: 0 } },
-          { $unset: { 'keyAltNames': '' }, $currentDate: { 'updateDate': true } },
-          { returnDocument: 'before' }
-        ]);
-        expect(result).to.deep.equal(r2.value);
+        expect(result).to.deep.equal(r.value);
       });
       it('reads keyAltNames and keyMaterial from DataKeyEncryptionKeyOptions', async() => {
```

Comment:
Not related - but I don't think this test tests `removeKeyAlternateName` implementation, or?

Reply:
Right, these should not be grouped together :+1: